addon/commands/insert-newLi-command.ts

```diff
@@ -17,261 +17,334 @@ export default class InsertNewLiCommand extends Command {
-        deleteNode = false;
-      } else {
-        return;
+      else if (direction == "left" && searchVar.previousSibling) {
+        searchVar = searchVar.previousSibling;
+        return searchVar;
+      }
+      else if (searchVar.parent) {
+        searchVar = searchVar.parent;
       }
     }
+    return null;
   }
 
-  private deleteSelection(first: ModelNode, last: ModelNode) {
-    //make sure the first node is deleted
-    let node = last;
-    let deleteNode = true;
-    while (node) {
-      //stop if we reached the first node
-      if (node == first) {
-        break;
+  // core algorithm
+  // this basically splits a li
+  // like so:
+  //
+  // <li>
+  //  <div class="some stuff">
+  //   some text
+  //       ^cursor
+  //  </div>
+  // </li>
+  //
+  // turns into:
+  //
+  // <li>
+  //  <div class="some stuff">
+  //   some
+  //  </div>
+  // </li>
+  // <li>
+  //  <div class="some stuff">
+  //   text
+  //  ^cursor
+  //  </div>
+  // </li>
+  //
+  // when a non-collapsed selection is preset
+  // it basically deletes all the highlighted stuff
+  // and turns it into a collapsed case
+  // if stuff is unclear please ask sergey.andreev@redpencil.io
+  insertNewLi(position: ModelPosition): void {
+
+    //split the nodes at position
+    position.split();
+
+    //variable initialization
+    let parentLi;
+    let parentUl;
+    let parentLiPos;
+    let toBeInserted: ModelNode[] = [];
+    let node = position.nodeAfter();
+
+    const newLiNodes = [];
+    const newLi = new ModelElement('li');
+
+    //try to find the first node we can copy if there is no nodeAfter
+    //looks for a right sibling of a direct ancestor untill we reach the parent li
+    if (!node) {
+      let searchVar = position.nodeBefore();
+      if (searchVar && searchVar.parent) {
+        searchVar = searchVar.parent;
       }
-      if (deleteNode) {
-        node.parent?.removeChild(node);
+      while (searchVar) {
+        if (ModelNode.isModelElement(searchVar) && searchVar.type === "li") {
+          break;
+        }
+        else if (searchVar.nextSibling) {
+          searchVar = searchVar.nextSibling;
+          node = searchVar;
+          break;
+        }
+        else if (searchVar.parent) {
+          searchVar = searchVar.parent;
+        }
       }
-      //remove all siblings to the left but keep direct parents
-      if (node.previousSibling) {
-        node = node.previousSibling;
-        deleteNode = true;
-      } else if (node.parent) {
-        node = node.parent;
-        deleteNode = false;
-      } else {
-        break;
+
+      //assume that the cursor is at the end
+      if (!node) {
+        node = position.nodeBefore();
+        if (!node) {
+          throw new Error("couldn't find a node to work with");
+        }
+
+        //find parent li and ul/ol as well as parent li position
+        parentLi = node.findAncestor(node => ModelNode.isModelElement(node) && (node.type === 'li'), false);
```

Comment:
I think I'll end up undeprecating findAncestor for the time being
it seems too popular